File: packages/python/infermap/infermap/domain_pack.py
"""DomainPackTarget — adapts a goldencheck-types DomainPack as an InferMap target."""
from __future__ import annotations

from goldencheck_types import DomainPack

from infermap.types import FieldInfo, SchemaInfo


class DomainPackTarget:
    """Wraps a goldencheck-types DomainPack as an InferMap target schema.

    Each canonical type in the pack becomes a target field. The type's
    ``name_hints`` populate ``sample_values`` so InferMap's existing
    ``FuzzyNameScorer`` can fire on column-name matches without code changes.
    """

    def __init__(self, pack: DomainPack):
        self.pack = pack

    def to_schema_info(self) -> SchemaInfo:
        fields = []
        for type_name, spec in self.pack.types.items():
            fields.append(
                FieldInfo(
                    name=type_name,
                    dtype="string",
                    sample_values=list(spec.name_hints),
                    metadata={
                        "value_signals": dict(spec.value_signals),
                        "confidence_threshold": spec.confidence_threshold,
                        "domain": self.pack.name,
                    },
                )
            )
        return SchemaInfo(
            fields=fields,
            source_name=f"domain:{self.pack.name}",
        )

File: packages/typescript/infermap/src/core/domainPack.ts
// DomainPackTarget — adapts a goldencheck-types DomainPack as an InferMap target.
import type { DomainPack } from "goldencheck-types";
import type { FieldInfo, SchemaInfo } from "./types.js";

export class DomainPackTarget {
  constructor(public readonly pack: DomainPack) {}

  toSchemaInfo(): SchemaInfo {
    const fields: FieldInfo[] = Object.entries(this.pack.types).map(
      ([type_name, spec]) => ({
        name: type_name,
        dtype: "string",
        sampleValues: [...spec.name_hints],
        nullRate: 0,
        uniqueRate: 0,
        valueCount: 0,
        metadata: {
          value_signals: spec.value_signals,
          confidence_threshold: spec.confidence_threshold,
          // Forward suppress + description so downstream scanners can read
          // them without re-loading the YAML pack. (Same shape as the
          // Python `DomainPackTarget.to_schema_info`.)
          suppress: [...spec.suppress],
          description: spec.description,
          domain: this.pack.name,
        },
      }),
    );
    return {
      sourceName: `domain:${this.pack.name}`,
      fields,
      requiredFields: [],
    };
  }
}

export function isDomainPackTarget(x: unknown): x is DomainPackTarget {
  return x instanceof DomainPackTarget;
}
